engine/table/src/main/java/io/deephaven/engine/table/impl/dataindex/TableBackedDataIndex.java

```diff
@@ -52,11 +52,17 @@ public class TableBackedDataIndex extends AbstractDataIndex {
     public TableBackedDataIndex(
             @NotNull final QueryTable sourceTable,
             @NotNull final String... keyColumnNames) {
-        this.keyColumnNames = List.of(keyColumnNames);
+        this(sourceTable, List.of(keyColumnNames));
+    }
+
+    public TableBackedDataIndex(
+            @NotNull final QueryTable sourceTable,
+            @NotNull final List<String> keyColumnNames) {
```

Comment:
Take `Collection` instead and make a defensive copy. Maybe actually better to just keep the array constructor.